src/iccpd/src/mlacp_sync_update.c

```diff
@@ -0,0 +1,659 @@
+int mlacp_fsm_update_mac_info_from_peer(struct CSM* csm, struct mLACPMACInfoTLV* tlv)
+{
+    struct Msg* msg = NULL;
+    struct MACMsg *mac_msg = NULL, mac_data;
+    struct LocalInterface* local_if = NULL;
+    uint8_t from_mclag_intf = 0;/*0: orphan port, 1: MCLAG port*/
+
+    if (!csm || !tlv)
+        return -255;
+
+#if 1
+    ICCPD_LOG_INFO(__FUNCTION__,
+                   "Received MAC Info, itf=[%s] vid[%d] MAC[%s]  type %d ",
+                   tlv->ifname, ntohs(tlv->vid), tlv->mac_str, tlv->type);
+#endif
+
+    /*Find the interface in MCLAG interface list*/
+    LIST_FOREACH(local_if, &(MLACP(csm).lif_list), mlacp_next)
+    {
+        if (local_if->type == IF_T_PORT_CHANNEL && strcmp(local_if->name, tlv->ifname) == 0)
+        {
+            from_mclag_intf = 1;
+            break;
+        }
+    }
+
+    if (!local_if)
+    {
+        if (!(local_if = local_if_find_by_name(tlv->ifname)))
+        {
+            /*If the interface is not exist, return*/
+            return -255;
+        }
+    }
+
+    /* update MAC list*/
+    TAILQ_FOREACH(msg, &(MLACP(csm).mac_list), tail)
+    {
+        mac_msg = (struct MACMsg*)msg->buf;
+
+        /*Same MAC is exist in local switch*/
+        if (strcmp(mac_msg->mac_str, tlv->mac_str) == 0 && mac_msg->vid == ntohs(tlv->vid))
+        {
+            if (tlv->type == MAC_SYNC_ADD)
+            {
+                mac_msg->age_flag &= ~MAC_AGE_PEER;
+                ICCPD_LOG_DEBUG(__FUNCTION__, "Recv ADD, Remove peer age flag:%d ifname  %s, add %s vlan-id %d, op_type %d",
+                                mac_msg->age_flag, mac_msg->ifname, mac_msg->mac_str, mac_msg->vid, mac_msg->op_type);
+
+                /*mac_msg->fdb_type = tlv->fdb_type;*/
+                /*The port ifname is different to the local item*/
+                if (strcmp(mac_msg->ifname, tlv->ifname) != 0 || strcmp(mac_msg->origin_ifname, tlv->ifname) != 0)
+                {
+                    if (mac_msg->fdb_type != MAC_TYPE_STATIC)
+                    {
+                        /*Update local item*/
+                        memcpy(&mac_msg->ifname, tlv->ifname, MAX_L_PORT_NAME);
+                        memcpy(&mac_msg->origin_ifname, tlv->ifname, MAX_L_PORT_NAME);
+                    }
+
+                    /*If the MAC is learned from orphan port, or from MCLAG port but the local port is down*/
+                    if (from_mclag_intf == 0 || (local_if->state == PORT_STATE_DOWN && strcmp(mac_msg->ifname, csm->peer_itf_name) != 0))
+                    {
+                        /*Redirect the mac to peer-link*/
+                        memcpy(&mac_msg->ifname, csm->peer_itf_name, IFNAMSIZ);
+
+                        /*Set MAC_AGE_LOCAL flag*/
+                        mac_msg->age_flag = set_mac_local_age_flag(csm, mac_msg, 1);
+
+                        /*Send mac add message to mclagsyncd*/
+                        add_mac_to_chip(mac_msg, MAC_TYPE_DYNAMIC);
+                    }
```

Comment:
This update might overwrite the local static MAC. There is a check just right above this code for static mac , can we please consider moving this code inside that block or update mac only if existing mac is not static.

Reply:
@Praveen-Brcm, Before ICCP submit to community, static MAC can not be configured, we can refine the MAC type later. Thanks.